src/Simulation/Simulators/OpenSystemsSimulator/Decompositions/IntrinsicInterface.qs

```diff
@@ -0,0 +1,210 @@
+                } apply {
+                    Native.CNOT(controls[0], target);
+                }
+            } elif Length(controls) == 1 {
+                // [Page 15 of arXiv:1206.0758v3](https://arxiv.org/pdf/1206.0758v3.pdf#page=15)
+                Adjoint T(controls[0]);
+                Adjoint T(controls[1]);
+                Controlled X([target], controls[0]);
+                T(controls[0]);
+                Controlled X([controls[1]], target);
+                Controlled X([controls[1]], controls[0]);
+                T(target);
+                Adjoint T(controls[0]);
+                Controlled X([controls[1]], target);
+                Controlled X([target], controls[0]);
+                Adjoint T(target);
+                T(controls[0]);
+                Controlled X([controls[1]], controls[0]);
+            } else {
+                ApplyWithLessControlsA(Controlled Z, (controls, target));
+            } 
+        }
+    }
+
+    operation S(target : Qubit) : Unit is Adj + Ctl {
+        body (...) {
+            Native.S(target);
+        }
+
+        controlled (controls, ...) {
+            if Length(controls) == 0 {
+                S(target);
+            } elif (Length(controls) == 1) {
+                T(controls[0]);
+                T(target);
+                within {
+                    Controlled X([controls[0]], target);
+                } apply {
+                    Adjoint T(target);
+                }
+            } else {
+                ApplyWithLessControlsA(Controlled S, (controls, target));
+            }
+        }
+    }
+
+    operation T(target : Qubit) : Unit is Adj + Ctl {
+        body (...) {
+            Native.T(target);
+        }
+
+        controlled (controls, ...) {
+            if Length(controls) == 0 {
+                T(target);
+            } else {
+                // TODO: Decompositions of `Controlled T` currently used in Q#
+                //       target packages rely on R1Frac, which is not yet
+                //       implemented in experimental simulation.
+                fail "Controlled T operations are not yet supported.";
+            }
+        }
+    }
+
+    // NB: We separate out this operation to avoid hardware targeting rewrite
+    //     steps from trying to lift this operation and modifying the C#
+    //     code gen in the process.
+    internal operation MeasureWithoutPauliI(bases : Pauli[], register : Qubit[]) : Result {
+        mutable newBases = [];
+        mutable newQubits = [];
+        // NB: using Zipped would be nice here, but this is built before
+        //     M.Q.Standard...
+        for idxBasis in 0..Length(bases) - 1 {
+            if bases[idxBasis] != PauliI {
+                set newBases += [bases[idxBasis]];
+                set newQubits += [register[idxBasis]];
+            }
+        }
+
+        if Length(newBases) == 0 {
+            return Zero;
+        } else {
+            return Measure(newBases, register);
```

Comment:
I think this needs to be `Measure(newBases, newQubits)` to ensure the matching sub-register is used.

Reply:
Yes, thank you for catching that!